modules/analytics-hub/variables.tf

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "description" {
+  description = "Resource description for data exchange."
+  default     = null
+  type        = string
+}
+
+variable "documentation" {
+  description = "Documentation describing the data exchange."
+  default     = null
+  type        = string
+}
+
+variable "factories_config" {
+  description = "Paths to data files and folders that enable factory functionality."
+  type = object({
+    listings = optional(string)
+  })
+  nullable = false
+  default  = {}
+}
+
+variable "iam" {
+  description = "Authoritative IAM bindings for the data exchange in {ROLE => [MEMBERS]} format."
+  type        = map(list(string))
+  default     = {}
+  nullable    = false
+}
+
+variable "icon" {
+  description = "Base64 encoded image representing the data exchange."
+  default     = null
+  type        = string
+}
+
+variable "listings" {
+  description = "Listings definitions in the form {LISTING_ID => LISTING_CONFIGS}. LISTING_ID must contain only Unicode letters, numbers (0-9), underscores (_). Should not use characters that require URL-escaping, or characters outside of ASCII, spaces."
+  type = map(object({
+    description      = optional(string)
```

Comment:
nit: can you put the non-optional fields first, then the optionals, then the object-typed ones; then sort each group alphabetically?

Reply:
done